src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Extensions/RazorCohostRequestContextExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor.Cohost;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.LanguageServerClient.Razor.Cohost;
+using StreamJsonRpc;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Extensions;
+
+internal static class RazorCohostRequestContextExtensions
+{
+    public static async Task<TResponse?> DelegateRequestAsync<IDelegatedParams, TResponse>(this RazorCohostRequestContext requestContext, string target, IDelegatedParams @params, ILogger logger, CancellationToken cancellationToken)
```

Comment:
Naming: `IDelegatedParams` should be `TDelegatedParams`

(though I'm not sure it really matters, `@params` could just be `object`)